File: packages/core-java/src/main/java/com/openpasskey/core/AuthenticationResult.java
package com.openpasskey.core;

/**
 * Result of successful WebAuthn authentication verification.
 */
public class AuthenticationResult {
    private final int signCount;
    private final int flags;
    private final boolean backupEligible;
    private final boolean backupState;

    public AuthenticationResult(int signCount, int flags, boolean backupEligible, boolean backupState) {
        this.signCount = signCount;
        this.flags = flags;
        this.backupEligible = backupEligible;
        this.backupState = backupState;
    }

    public int getSignCount() { return signCount; }
    public int getFlags() { return flags; }
    public boolean isBackupEligible() { return backupEligible; }
    public boolean isBackupState() { return backupState; }
}

File: packages/core-ruby/lib/open_passkey/authentication_result.rb
# frozen_string_literal: true

module OpenPasskey
  class AuthenticationResult
    attr_reader :sign_count, :flags, :backup_eligible, :backup_state

    def initialize(sign_count:, flags:, backup_eligible:, backup_state:)
      @sign_count = sign_count
      @flags = flags
      @backup_eligible = backup_eligible
      @backup_state = backup_state
    end
  end
end
